.github/workflows/generate-and-build-sdks.yml

```diff
@@ -84,6 +90,39 @@ jobs:
             source/*
             !source/src/*.o
 
+  build-java-sdk:
+    name: Build Java SDK
+    runs-on: ubuntu-latest
+    needs: generate-sdk-sources
+    steps:
+      - name: Install dependencies
+        run: sudo apt-get install maven
+
+      - name: Retrieve Java SDK source
+        uses: actions/download-artifact@v4
+        with:
+          name: SDK_Source_Java
+          path: source/
+
+      - name: Set up JDK 21
```

Comment:
Shouldn't we be using what we specify in the pom, i.e. 11. Unless you're implying we should upgrade, but if we are to believe this [article](https://newrelic.com/resources/report/2024-state-of-the-java-ecosystem) the number of people on 21 is still small (1.4%) compared to 11 (32.9%) and 17 (35.4%).

Reply:
right so that would be fine for the xapi jar, but turns out the javadoc plugin doesn't support some of the new javadoc I've added when using JDK11, so the build will fail (unless we skip building the docs)

I could do JDK 17 too, but thought it'd be better to just do JDK 21

honestly shouldn't be much of an issue since we're not actually shipping this, and if it builds for JDK 21 it'll likely build for JDK 17

Reply:
see https://github.com/danilo-delbusso/xen-api/actions/runs/10788472033/job/29919633081 for the failed run (was using JDK 11)

Reply:
Hmm, the failures look like things that can be disabled by parameters in the pom, see here https://stackoverflow.com/questions/39616344/how-to-disable-javadoc-warnings-in-maven-javadoc-plugin

Reply:
I'm not worried about the "missing" bit, those are fairly harmless

my issue is with these:

```
/home/runner/work/xen-api/xen-api/source/xen-api/src/main/java/com/xensource/xenapi/Types.java:14404: error: type arguments not allowed here
Error:      * @param object The {@link Object} to be converted to a {@link Map<GPUGroup, GPUGroup.Record>} object.
Error:                                                                     ^
Error:  /home/runner/work/xen-api/xen-api/source/xen-api/src/main/java/com/xensource/xenapi/Types.java:14405: error: type arguments not allowed here
Error:      * @return A {@link Map<GPUGroup, GPUGroup.Record>} object created from the input {@link Object}'s {@link String} representation,
Error:                         ^
Error:  /home/runner/work/xen-api/xen-api/source/xen-api/src/main/java/com/xensource/xenapi/Types.java:14426: error: type arguments not allowed here
Error:      * Converts an {@link Object} to a {@link Map<Long, Double>} object.
```

not sure if they'll be ok if I suppress them 🤔 I just thought it better to move to JDK 21

Reply:
that uses JDK17 (See Dockerfile in ci-containers) - I could switch to that here?

Reply:
tested here:
https://github.com/danilo-delbusso/xen-api/actions/runs/10826226513?pr=6